agent/stats/service_connect_linux.go

```diff
@@ -117,7 +117,14 @@ func convertToTACSStats(mf map[string]*prometheus.MetricFamily, taskId string) (
 					}
 					metricCounts = append(metricCounts, &metricCount)
 				}
-				metricCounts = convertHistogramMetricCounts(metricCounts)
+
+				metricValues, metricCounts = convertHistogramMetricCounts(metricValues, metricCounts)
+
+				// If all values are 0 in metricCount, then no need to send the metrics to TACS
+				if len(metricCounts) == 0 && len(metricValues) == 0 {
```

Comment:
nit 
- `metricCounts` and `metricValues` should have the same size so checking one of them is good enough.
- Can we add a debug log here to indicate that the metric has been skipped?

Reply:
Fixed the above 2.

Reply:
Instead of individually checking 0 values in Counts/Values array, can we just check the SampleCount for being 0 (`metric.Histogram.SampleCount` I think) on L99 and skip the processing of its buckets entirely?